### PR TITLE
ishd.c: Added freeaddrinfo to free addrs

### PR DESCRIPTION
**Pull request by the-c0d3r**
PR for #2.
May I know if there's anything else I can help improve this project? I'm very interested to add more things for this. This commit just adds one line. 

Anyway, thanks for giving me a chance.

### CODE REVIEW

**Comment by tedjp** [comment]
Awesome, thanks! And thanks for cleaning up that trailing whitespace I left in — oops.

There's one more leak I know of, described in #1 if you want to have a go at it.

Other ideas to do on this project:
- Debian/Ubuntu packaging. Something along the lines of [wtfd](https://github.com/tedjp/wtfd/tree/master/debian). Best to make the service as restricted as possible.
- Allocate a pty (pseudo-terminal) to enable job control in the shell. Start by reading the `openpty(3)` manpage.

But it's mainly a novelty so I don't think it's worth doing much :)

File: ishd.c [hunk 1]
```diff
@@ -132,5 +132,7 @@ int main(void) {
         return 1;
     }
 
+    freeaddrinfo(addrs);
```
Comment (tedjp) [inline ishd.c]: Consider moving the `freeaddrinfo()` call right after the `bind()` call and its error handling; the addresses are not used after that. This line is only reached after the main server loop ends (if `accept4()` fails).

**Comment by the-c0d3r** [comment]
Hi, thanks for the comments. I didn't catch that bind usage of `addrs`. It was late and I thought I just need to add it right before quitting. I'll take a look at the second idea that you listed. Please suggest me if you have any other projects which are beginner friendly. Thanks!

**Comment by tedjp** [comment]
My other projects that might be beginner-friendly. Most don't have any issues recorded in GitHub but maybe you'll run into some bugs if you try them.

* [wtfd](https://github.com/tedjp/wtfd) is very similar; it's another novelty network service. I'm not aware of any bugs in it but if you want to experiment with a simple non-forking, non-threaded network service it could get you started.
* [firehose](https://github.com/tedjp/firehose) IPv4 multicast forwarder. One big problem: [no IPv6 support](https://github.com/tedjp/firehose/issues/2).
* [stallone](https://github.com/tedjp/stallone) NAT port mapping client/server. This project is fairly old so it might have bit-rotted. There are probably some new compiler warnings from the last decade that ought to be resolved.

Depending on your level of experience it's worth reading [A Tutorial on Pointers and Arrays in C](http://pw1.netcom.com/~tjensen/ptr/pointers.htm) and [Beej's Guide to Network Programming](https://beej.us/guide/bgnet/).

**Pull request merged**